### DISCUSSION TITLE
Collection API Prior Art

### DISCUSSION THREAD

**Discussion by samdphillips** [racket/rhombus#401] [Ideas]
At our meeting today (2023/11/09), we discussed ways forward with Collections.  I think having a list of prior art could help guide where the design may go.  Please add links to any other APIs that you feel should be considered.

- ANSI Smalltalk collections: (Section 5.7) https://wiki.squeak.org/squeak/uploads/172/standard_v1_9-indexed.pdf
- Rust: https://doc.rust-lang.org/std/collections/index.html

Somehow, I missed that @samth posted a list of other Collection APIs after a September meeting.  Here are those links.
- Guava (@jackfirth's major inspiration): https://github.com/google/guava/wiki
- Scala: https://docs.scala-lang.org/overviews/collections-2.13/overview.html
- Java: https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/util/doc-files/coll-overview.html
- Kotlin: https://kotlinlang.org/docs/collections-overview.html
- OCaml Batteries: https://ocaml-batteries-team.github.io/batteries-included/hdoc2/
- OCaml Core: https://ocaml.org/p/core_kernel/latest/doc/index.html

**Comment by soegaard** [comment]
- Julia: https://docs.julialang.org/en/v1/base/arrays/#Views-(SubArrays-and-other-view-types)
- Julia: https://docs.julialang.org/en/v1/base/collections/
- Swift: https://developer.apple.com/documentation/swift/collection

**Comment by gfr1337** [comment]
Dylan: https://opendylan.org/books/drm/Collections_Overview

**Comment by sorawee** [comment]
Saw this on Twitter: https://stefan-marr.de/papers/px-marr-daloze-few-versatile-vs-many-specialized-collections/
